src/include/misc.h

```diff
@@ -39,24 +39,13 @@
         (v) = (val);        \
         (void)(v);          \
     } while (0);
-#define WT_IGNORE_RET(call)                \
-    do {                                   \
-        uintmax_t __ignored_ret;           \
-        __ignored_ret = (uintmax_t)(call); \
-        WT_UNUSED(__ignored_ret);          \
-    } while (0)
-#define WT_IGNORE_RET_BOOL(call)  \
-    do {                          \
-        bool __ignored_ret;       \
-        __ignored_ret = (call);   \
-        WT_UNUSED(__ignored_ret); \
-    } while (0)
-#define WT_IGNORE_RET_PTR(call)    \
-    do {                           \
-        const void *__ignored_ret; \
-        __ignored_ret = (call);    \
-        WT_UNUSED(__ignored_ret);  \
-    } while (0)
+
+/* 
+ * Explicitly suppress unused return value warning at function call sites. 
+ */
+#define WT_IGNORE_RET(call) ((void)!(call))
+#define WT_IGNORE_RET_BOOL(call) ((void)!(call))
```

Comment:
Can we merge these macros into a single `WT_IGNORE_RET`?
We only have 7 instances of `WT_IGNORE_RET_BOOL` and `WT_IGNORE_RET_PTR` in `src` so the diff will be small.